docker-entrypoint.sh

```diff
@@ -9,15 +9,21 @@ if [ "${1:0:1}" = '-' ]; then
     set -- atlantis "$@"
 fi
 
-# Look for atlantis subcommands.
-if atlantis --help "$1" 2>&1 | grep -q "atlantis $1"; then
-    # We can't use the return code to check for the existence of a subcommand, so
-    # we have to use grep to look for a pattern in the help output.
+# If the user is running an atlantis subcommand (ex. server) then we want to prepend
+# atlantis as the first arg to exec. To detect if they're running a subcommand
+# we take the potential subcommand and run it through atlantis help {subcommand}.
+# If the output contains "atlantis subcommand" then we know it's a subcommand
+# since the help output contains that string. For anything else (ex. sh)
+# it won't contain that string.
+# NOTE: We use grep instead of the exit code since help always returns 0.
+if atlantis help "$1" 2>&1 | grep -q "atlantis $1"; then
     set -- atlantis "$@"
 fi
 
-# If we are running atlantis, make sure it executes as the proper user.
-if [ "$1" = 'atlantis' ]; then
+# If we're running as root and we're trying to execute atlantis then we use
+# gosu to step down from root and run as the atlantis user.
+# In OpenShift, containers are run as a random users so we don't need to use gosu.
+if [[ $(id -u) == 0 ]] && [[ "$1" = 'atlantis' ]]; then
```

Comment:
I also needed to add a user in /etc/passwd as you can see in https://github.com/runatlantis/atlantis/pull/346. Otherwise terraform was not able to clone a module by git.

Reply:
oh interesting. okay I've added that as well.